File: clojure/cljsearch/test/cljsearch/filetypes_test.clj
(ns cljsearch.file-types-test
  (:require [clojure.test :refer :all])
  (:use [cljsearch.filetypes :only (get-file-type)]))

(deftest test-get-file-type-archive-file
  (let [f "archive.zip"]
    (testing "test-get-file-type-archive-file"
      (is (= (get-file-type f) :archive)))))

(deftest test-get-file-type-binary-file
  (let [f "binary.exe"]
    (testing "test-get-file-type-binary-file"
      (is (= (get-file-type f) :binary)))))

(deftest test-get-file-type-text-file
  (let [f "text.txt"]
    (testing "test-get-file-type-text-file"
      (is (= (get-file-type f) :text)))))

(deftest test-get-file-type-code-file
  (let [f "code.clj"]
    (testing "test-get-file-type-code-file"
      (is (= (get-file-type f) :code)))))

(deftest test-get-file-type-xml-file
  (let [f "markup.xml"]
    (testing "test-get-file-type-xml-file"
      (is (= (get-file-type f) :xml)))))

(deftest test-get-file-type-unknown-file
  (let [f "unknown.xyz"]
    (testing "test-get-file-type-unknown-file"
      (is (= (get-file-type f) :unknown)))))

File: typescript/tssearch/tests/filetypes.test.ts
/*
 * filetypes.test.js
 *
 * Some tests of filetypes.js
 */

import {FileType} from '../src/filetype';
import {FileTypes} from '../src/filetypes';

describe('testing filetypes', () => {
    it('testFileTypesArchiveFile', () => {
        const filename = 'archive.zip';
        const res: boolean = FileTypes.isArchiveFile(filename);
        expect(res).toBeTruthy();

        const type: FileType = FileTypes.getFileType(filename);
        expect(type).toBe(FileType.Archive);
    });

    it('testFileTypesBinaryFile', () => {
        const filename = 'binary.exe';
        const res: boolean = FileTypes.isBinaryFile(filename);
        expect(res).toBeTruthy();

        const type: FileType = FileTypes.getFileType(filename);
        expect(type).toBe(FileType.Binary);
    });

    it('testFileTypesCodeFile', () => {
        const filename = 'code.js';
        const res: boolean = FileTypes.isCodeFile(filename);
        expect(res).toBeTruthy();

        const type: FileType = FileTypes.getFileType(filename);
        expect(type).toBe(FileType.Code);
    });

    it('testFileTypesTextFile', () => {
        const filename = 'text.txt';
        const res: boolean = FileTypes.isTextFile(filename);
        expect(res).toBeTruthy();

        const type: FileType = FileTypes.getFileType(filename);
        expect(type).toBe(FileType.Text);
    });

    it('testFileTypesXmlFile', () => {
        const filename = 'markup.xml';
        const res: boolean = FileTypes.isXmlFile(filename);
        expect(res).toBeTruthy();

        const type: FileType = FileTypes.getFileType(filename);
        expect(type).toBe(FileType.Xml);
    });

    it('testFileTypesUnknownFile', () => {
        const filename = 'unknown.xyz';
        const res: boolean = FileTypes.isUnknownFile(filename);
        expect(res).toBeTruthy();

        const type: FileType = FileTypes.getFileType(filename);
        expect(type).toBe(FileType.Unknown);
    });
});
